src/Stratis.Bitcoin.IntegrationTests/SegWitTests.cs

```diff
@@ -39,35 +40,63 @@ public void TestSegwitActivation()
-                IConsensusLoop consensusLoop = node1.FullNode.NodeService<IConsensusLoop>();
-                ThresholdState[] bip9State = consensusLoop.NodeDeployments.BIP9.GetStates(node1.FullNode.Chain.Tip.Previous);
+                try
+                {
+                    // generate 450 blocks, block432 is a segwit block
+                    coreRpc.Generate(450);
 
-                Money amount = new Money(5.0m, MoneyUnit.BTC);
-                HdAddress destination1 = wm1.GetUnusedAddress(new WalletAccountReference("alice1", "account 0"));
+                    TestHelper.WaitLoop(() => stratisNode.CreateRPCClient().GetBestBlockHash() == coreNode.CreateRPCClient().GetBestBlockHash());
 
-                coreRpc.SendToAddress(BitcoinAddress.Create(destination1.Address, Network.RegTest), amount);
+                    // segwit activation on Bitcoin regtest
+                    // - On regtest deployment state changes every 144 block, the threshold for activating a rule is 108 blocks.
+                    // segwit should be:
+                    // - Defined up to block 143
+                    // - Started at block 144 to block 288 
+                    // - LockedIn 288 (as segwit blocks should already be signaled)
+                    // - Active at block 432 
 
-                coreRpc.Generate(1);
+                    IConsensusLoop consensusLoop = stratisNode.FullNode.NodeService<IConsensusLoop>();
+                    ThresholdState[] segwitDefinedState = consensusLoop.NodeDeployments.BIP9.GetStates(stratisNode.FullNode.Chain.GetBlock(142));
+                    ThresholdState[] segwitStartedState = consensusLoop.NodeDeployments.BIP9.GetStates(stratisNode.FullNode.Chain.GetBlock(143));
+                    ThresholdState[] segwitLockedInState = consensusLoop.NodeDeployments.BIP9.GetStates(stratisNode.FullNode.Chain.GetBlock(287));
+                    ThresholdState[] segwitActiveState = consensusLoop.NodeDeployments.BIP9.GetStates(stratisNode.FullNode.Chain.GetBlock(431));
 
-                Assert.Equal(ThresholdState.Active, bip9State.GetValue((int)BIP9Deployments.Segwit));
+                    // check that segwit is got activated at block 432
+                    Assert.Equal(ThresholdState.Defined, segwitDefinedState.GetValue((int)BIP9Deployments.Segwit));
+                    Assert.Equal(ThresholdState.Started, segwitStartedState.GetValue((int)BIP9Deployments.Segwit));
+                    Assert.Equal(ThresholdState.LockedIn, segwitLockedInState.GetValue((int)BIP9Deployments.Segwit));
+                    Assert.Equal(ThresholdState.Active, segwitActiveState.GetValue((int)BIP9Deployments.Segwit));
+                }
+                finally
+                {
+                    Network.RegTest.Consensus.BIP9Deployments[BIP9Deployments.Segwit] = prevSegwitDeployment;
+                }
             }
         }
+
+        public void TestSegwit_MinedOnStratisNode_ActivatedOn_Corenode()
```

Comment:
i guess this should be private or `[Fact]` to avoid warning

Reply:
ah yeah good point